src/Functions/multiIf.cpp

```diff
@@ -117,6 +118,15 @@ class FunctionMultiIf final : public FunctionIfBase
             types_of_branches.emplace_back(arg);
         });
 
+        if (context->getSettingsRef().allow_experimental_variant_type && context->getSettingsRef().use_variant_when_no_common_type_in_if)
```

Comment:
FYI we do some optimizations with if/multiIf (e.g. `IfChainToMultiIfPass` and `MultiIfToIfPass`) which could change the return type after those optimizations, it would be nice to test those optimizations with enabled analyzer to see everything is okay

Reply:
Sure, I will test

Reply:
Yeah, there was a small problem with `IfChainToMultiIfPass ` with analyzer with returned types, we were trying to create nested `Variant` types (but it's not allowed). I just changed the logic of creating result `Variant` type in `if/multiIf` so now it works (and I think it's now better than it was).